### ISSUE TITLE
Support TMS?

### ISSUE THREAD

**Issue by jleppert** [mapbox/mapbox-gl-js#1616]
Adding scheme: tms to the layer source does nothing. Is it possible to get this supported? Can someone point me to the direction in the code where tile requests are done?

Thanks,

Johnathan

**Comment by bhousel** [comment]
This is supported - you can specify your tile source as `type: raster` and embed TileJSON properties directly in the style, like this:

```
    "sources": {
        "example_id": {
            "type": "raster",
            "tiles": [
                "http://a.example.com/{z}/{x}/{y}.png",
                "http://b.example.com/{z}/{x}/{y}.png"
            ],
            "tileSize": 256
        }
    },
```

see also [Style Spec - sources](https://www.mapbox.com/mapbox-gl-style-spec/#sources)

**Comment by tmcw** [comment]
That doesn't mean that we support the TMS scheme, though, right? TMS is flipped from the OSM/XYZ scheme.

I'm not sure we _want_ to support this in core: TMS is more or less a mistake of history and rare as a result.

**Comment by bhousel** [comment]
Sorry, read this too quickly, reopening.  I didn't realize it was a subset of TileJSON that we are missing support for..

**Comment by jleppert** [comment]
Yeah, I understand it's not supported although the TileJSON specs seem to imply that it would work (see the schema definition).

**Comment by rm3dbeckers** [comment]
I'm having the same problem: I can set "scheme": "tms", but it would still behave as if I had left it at "xyz".

**Comment by tmcw** [comment]
This is a known issue: as discussed earlier, Mapbox GL JS does not support TMS URLs yet. PRs accepted, otherwise please expect this to not work yet, because it isn't implemented yet.

**Comment by lucaswoj** [comment]
Resolved by #2565
